spec/abci/abci++_app_requirements.md

```diff
@@ -5,6 +5,32 @@ title: Requirements for the Application
 
 # Requirements for the Application
 
+- [Formal Requirements](#formal-requirements)
+- [Managing the Application state and related topics](#managing-the-application-state-and-related-topics)
+  - [Connection State](#connection-state)
+    - [Concurrency](#concurrency)
+    - [Finalize Block](#finalizeblock)
+    - [Commit](#commit)
+    - [Candidate States](#candidate-states)
+  - [States and ABCI++ Connections](#states-and-abci%2B%2B-connections) 
+    - [Consensus Connection](#consensus-connection)
+    - [Mempool Connection](#mempool-connection)
+    - [Info/Query Connection](#infoquery-connection)
+    - [Snapshot Connection](#snapshot-connection)
+  - [Transaction Results](#transaction-results)
+  - [Updating the Validator Set](#updating-the-validator-set)
+  - [Consensus Parameters](#consensus-parameters)
+    - [List of Parameters](#list-of-parameters)
+    - [Updating Consensus Parameters](#updating-consensus-parameters)
+  - [Query](#query)
+    - [Query Proofs](#query-proofs)
+    - [Peer Filtering](#peer-filtering)
+    - [Paths](#paths)
+  - [Crash Recovery](#crash-recovery)
+  - [State Sync](#state-sync)
+- [Application configuration required to switch to ABCI2.0](#application-configuration-required-to-switch-to-abci-20)
+
+
```

Comment:
I [suggested](https://github.com/cometbft/cometbft/pull/484/files#r1141823124) to add a TOC, because I didn't realize GH has one. I managed to find it in the left-hand side of the view:

<img width="1282" alt="Screenshot 2023-03-21 at 09 33 49" src="https://user-images.githubusercontent.com/1757002/226553670-d86cfec3-0273-49ee-8b1f-9be661ab6a48.png"> 

Thanks for highlighting that Lasaro! I still think having the TOC directly in the file does not hurt.

> something to think about going forward when writing specs, RFC and ADR.

Agree. We might want to have a consistent approach. Good discussion to have internally another time.